### PR TITLE
[FIX] sale,purchase,account: Duplicating document_tax_totals

### PR DESCRIPTION
**Pull request by anko-odoo**
If user would add addictional field in one of reports it could break other reports as other models do not share all the same fields. Now modules inherit template and create their version of field.

task-2858097

Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Review by Demesmaeker — COMMENTED** [review]
Hello,

Be aware that in the modules `sale` and `purchase`, you also have a `(*_)portal_templates` that now t-call the `account.document_tax_totals`, this should probably be changed too =]

File: addons/purchase/views/portal_templates.xml [hunk 1]
```diff
@@ -374,7 +374,7 @@
   <template id="purchase_order_portal_content_totals_table">
       <table class="table table-sm">
           <t t-set="tax_totals" t-value="order.tax_totals"/>
-          <t t-call="account.document_tax_totals"/>
+          <t t-call="purchase.document_tax_totals"/>
```
Comment (Demesmaeker) [inline addons/purchase/views/portal_templates.xml:377]: Thinking of it... Are we sure we want the fields added on studio to be on the portal too? :thinking: 
But then, if we change it in studio while still calling the initial template, changes will be applied on every report too, which we certainly don't want either... Maybe ask @dalaOdoo what he thought of it, I'm not totally sure what can/can't be done in studio... :thinking:

**Review by Demesmaeker — COMMENTED** [review]
:duck: More food for thoughts than anything else

File: addons/purchase/views/portal_templates.xml (hunk 1 above)
Comment (anko-odoo) [reply addons/purchase/views/portal_templates.xml:377]: @Demesmaeker I asked, and from what I understand it is supposed to be like this, that when you change in the report it also changes in portal view.

File: addons/purchase/report/purchase_order_templates.xml [hunk 2]
```diff
@@ -135,4 +135,7 @@
         </t>
     </t>
 </template>
+
+<template id="document_tax_totals" inherit_id="account.document_tax_totals_template" primary="True"></template>
```
Comment (Demesmaeker) [inline addons/purchase/report/purchase_order_templates.xml:139]: You can take off the `primary=True` here too then ^^

**Review by Demesmaeker — COMMENTED** [review]
:duck:

**Review by Demesmaeker — APPROVED** [review]

File: addons/account/views/report_invoice.xml [hunk 3]
```diff
@@ -291,6 +291,8 @@
             </tr>
         </template>
 
+        <template id="document_tax_totals" inherit_id="account.document_tax_totals_template" primary="True"></template>
```
[the remark this answers is not in the document]
Comment (Demesmaeker) [reply addons/account/views/report_invoice.xml:302]: Edit: We finally saw that `primary=True` is also needed in order to be able to use Studio with the report, so we need to keep this

File: addons/purchase/report/purchase_order_templates.xml (hunk 2 above)
Comment (Demesmaeker) [reply addons/purchase/report/purchase_order_templates.xml:139]: Edit: see https://github.com/odoo/odoo/pull/106140#discussion_r1081495797
We kept the primary in the end

File: addons/account/views/report_invoice.xml [hunk 4]
```diff
@@ -299,6 +299,11 @@
             </tr>
         </template>
 
+        <!-- Empty template that copies document_tax_totals_template, so that when it's edited in one
+             module it only edits its copy not original, which would cause an error if added field 
+             does not exists in other modules-->
```
Comment (Demesmaeker) [inline addons/account/views/report_invoice.xml:304]: This comment can be shorter and still understandable.
```suggestion
        <!-- Allow edits (e.g. studio) without changing the often inherited base template -->
```

**Review by Demesmaeker — APPROVED** [review]
Hello, you can add the same comment on the different modules =]

**Review by Feyensv — APPROVED** [review]
@robodoo r+

**Pull request closed without merging**